app/models/ability.rb

```diff
@@ -0,0 +1,34 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    user ||= User.new # guest user (not logged in)
+    if user.admin?
+      can :manage, :all
+    else
+      can :manage, Post, author: user
+      can :manage, Comment, author: user
+      can :read, :all
+    end
+    #
+    # The first argument to `can` is the action you are giving the user
+    # permission to do.
+    # If you pass :manage it will apply to every action. Other common actions
+    # here are :read, :create, :update and :destroy.
+    #
+    # The second argument is the resource the user can perform the action on.
+    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, :published => true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+  end
+end
```

Comment:
- [ ] I highlighted this because I couldn't find signup codes but I am not able to sign up it always shows a password error though I filled it please check what is causing this because I can't check authorization rules without signing up or if I am missing out something please add in the README each and every step
![d](https://user-images.githubusercontent.com/44745855/147371512-964c3691-b370-441e-a730-192fef175fb8.png)